src/Illuminate/Cache/DatabaseLock.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace Illuminate\Cache;
+
+use Illuminate\Database\Connection;
+use Illuminate\Database\QueryException;
+
+class DatabaseLock extends Lock
+{
+    /**
+     * The database connection instance.
+     *
+     * @var \Illuminate\Database\Connection
+     */
+    protected $connection;
+
+    /**
+     * The database table name.
+     *
+     * @var string
+     */
+    protected $table;
+
+    /**
+     * The prune probability odds.
+     *
+     * @var string
```

Comment:
You define it as string, but describe and default it to an array of ints 🤔

Reply:
I wish Laravel lead devs would start thinking about a future with static analysers and more type declarations

Reply:
As soon as we have PHP 8 as the minimum version which has union type support we could *start* considering it. But that'd be September 2022 at the very earliest: https://www.php.net/supported-versions.php